### PR TITLE
connect maxHeartbeats in CoreM to kernel variable max_heartbeats

### PR DESCRIPTION
**Pull request by bollu**

### CODE REVIEW

**Comment by Kha** [comment]
The basic approach looks good (assuming it works :) ). Is there anything left apart from cleaning up the PR?

File: src/Lean/MonadEnv.lean [hunk 1]
```diff
@@ -152,26 +152,26 @@ private def checkUnsupported [Monad m] [MonadEnv m] [MonadError m] (decl : Decla
...
   | Except.ok env   => setEnv env
   | Except.error (KernelException.other msg) =>
     throwError msg
   | Except.error ex =>
     throwKernelException ex
 
-def addAndCompile [Monad m] [MonadEnv m] [MonadError m] [MonadOptions m] [MonadLog m] [AddMessageContext m] (decl : Declaration) : m Unit := do
-  addDecl decl;
-  compileDecl decl
+def addAndCompile [Monad m] [MonadEnv m] [MonadError m] [MonadOptions m] [MonadLog m] [AddMessageContext m] (maxHeartbeats: Nat) (decl : Declaration) : m Unit := do
```
Comment (gebner) [inline src/Lean/MonadEnv.lean:172]:
```suggestion
def addAndCompile (decl : Declaration) : CoreM Unit := do
```
I think we should avoid the extra argument here.  (And adding a `MonadHeartbeat` class feels overkill here, this isn't even a combinator.)

File: src/Lean/Meta/SynthInstance.lean [hunk 2]
```diff
@@ -159,7 +159,7 @@ structure TableEntry where
 
 structure Context where
   maxResultSize : Nat
-  maxHeartbeats : Nat
+  maxHeartbeats : Nat := 6543
```
Comment (gebner) [inline src/Lean/Meta/SynthInstance.lean:162]: Please revert.

**Comment by Kha** [comment]
For "quick guesses", `--output-on-failure` would have been helpful :) . And the CI won't run until you address the merge conflicts.

**Comment by leodemoura** [comment]
@bollu Are you still planning to work on this PR, or should I close it?

**Comment by bollu** [comment]
@leodemoura Closed to clear Lean's PR queue; I shall send a completed PR once the solution does not fail on seemingly unrelated test cases.

**Pull request closed without merging**